doctr/models/predictor/tensorflow.py

```diff
@@ -41,13 +44,20 @@ def __init__(
     def __call__(
         self,
         pages: List[Union[np.ndarray, tf.Tensor]],
+        straighten_pages: bool = False,
         **kwargs: Any,
     ) -> Document:
 
         # Dimension check
         if any(page.ndim != 3 for page in pages):
             raise ValueError("incorrect input shape: all pages are expected to be multi-channel 2D images.")
 
+        # Detect document rotation and rotate pages
+        if straighten_pages:
+            page_angles = []
+            page_orientations = [estimate_orientation(page) for page in pages]
+            pages = [rotate_image(page, angle) for page, angle in zip(pages, page_orientations)]
+
```

Comment:
We're still missing something to reproject the detections into the original image after `self._process_predictions`:
- right now if we merge, the pages will be rotated for inference
- but we need to map them back after inference

Reply:
Yes done !